src/sema/yul/ast.rs

```diff
@@ -12,6 +12,7 @@ use std::sync::Arc;
 #[derive(Debug, Clone)]
 pub struct InlineAssembly {
     pub loc: pt::Loc,
+    pub memory_safe: bool,
```

Comment:
I'm a bit confused. We add this `memory_safe` field here. But I never see this being actually used. So why bother adding it?

Alternatively we should have a comment explaining what's going on here

Reply:
I've added a comment to this struct.

Do you think this clear enough?

Reply:
Whlie that certainly helps, I meant that the comment should mention as to why this field is present but not (yet?) used any further

Reply:
Good point. I've explained it a bit more.